python/taichi/lang/__init__.py

```diff
@@ -188,29 +197,33 @@ def _get_or_make_arch_checkers(kwargs):
 
 # test with all archs
 def all_archs_with(**kwargs):
-  def decorator(func):
-    def test(*func_args, **func_kwargs):
+  kwargs = _deepcopy(kwargs)
+  def decorator(test):
+    def wrapped(*test_args, **test_kwargs):
       import taichi as ti
-      can_run_on = func_kwargs.pop(
+      can_run_on = test_kwargs.pop(
           _tests_arch_checkers_argname, _ArchCheckers())
       # Filter away archs that don't support 64-bit data.
-      fp = func_kwargs.get('default_fp', ti.get_runtime().default_fp)
-      ip = func_kwargs.get('default_ip', ti.get_runtime().default_ip)
+      fp = kwargs.get('default_fp', ti.f32)
```

Comment:
If `default_cfg()` is immutable, we can use something like `core.default_cfg().default_fp` here, and the default values will have a single source of truth..

Reply:
Yeah that would be a good temporary solution. Actually, I'm considering dropping the evil `ti.cfg`/`default_cfg` interface since it's too confusing.